lib/repository/online_graphql_repository/plan_graphql_model.dart

```diff
@@ -0,0 +1,217 @@
+import 'package:flutter/material.dart';
+import 'package:trufi_core/trufi_models.dart';
+
+enum TransportMode { walk, rail, bus, transit, car }
+
+extension TransportModeExtension on TransportMode {
+  static const names = {
+    TransportMode.walk: 'WALK',
+    TransportMode.bus: 'BUS',
+    TransportMode.car: 'CAR',
+    TransportMode.rail: 'LIGHT RAIL',
+    TransportMode.transit: 'TRANSIT',
+  };
+  String get name => names[this] ?? 'WALK';
+}
+
+const transportModeMap= {
+  'WALK': TransportMode.walk,
+  'BUS': TransportMode.bus,
+  'CAR': TransportMode.car,
+  'RAIL': TransportMode.rail,
+  'TRANSIT': TransportMode.transit,
+};
+
+TransportMode _getTransportMode(String enumText) 
+  => transportModeMap[enumText] ?? TransportMode.walk;
+
+class PlanGraphQl {
+  PlanGraphQl({
+    this.from,
+    this.to,
+    this.itineraries,
+  });
+
+  final _Location from;
+  final _Location to;
+  final List<_Itinerary> itineraries;
+
+  factory PlanGraphQl.fromJson(Map<String, dynamic> json) => PlanGraphQl(
+        from: _Location.fromJson(json["from"] as Map<String, dynamic>),
+        to: _Location.fromJson(json["to"] as Map<String, dynamic>),
+        itineraries: List<_Itinerary>.from(
+          (json["itineraries"] as List<dynamic>).map(
+            (x) => _Itinerary.fromJson(x as Map<String, dynamic>),
+          ),
+        ),
+      );
+
+  Map<String, dynamic> toJson() => {
+        "from": from.toJson(),
+        "to": to.toJson(),
+        "itineraries": List<dynamic>.from(itineraries.map((x) => x.toJson())),
+      };
+
+  Plan toPlan() {
+    return Plan(
+      to: to.toPlanLocation(),
+      from: from.toPlanLocation(),
+      itineraries: itineraries
+          .map(
+            (itinerary) => itinerary.toPlanItinerary(),
+          )
+          .toList(),
+      error: itineraries.isEmpty ? PlanError(404, 'Not found routes') : null,
+    );
+  }
+}
+
+class _Location {
```

Comment:
Can we make these parts reusable, I guess a Location will never be that unique and should be used at multiple parts, maybe we can even upgrade the TrufiLocation with a Location object.